lightning/src/ln/channel.rs

```diff
@@ -4027,7 +4028,8 @@ impl<Signer: Sign> Channel<Signer> {
 	/// In the first case, we store the confirmation height and calculating the short channel id.
 	/// In the second, we simply return an Err indicating we need to be force-closed now.
 	pub fn transactions_confirmed<L: Deref>(&mut self, block_hash: &BlockHash, height: u32, txdata: &TransactionData, logger: &L)
-			-> Result<Option<msgs::FundingLocked>, msgs::ErrorMessage> where L::Target: Logger {
+	-> Result<Option<msgs::FundingLocked>, (msgs::ErrorMessage, ClosureReason)>
```

Comment:
Rather than using a tuple for the error type, consider just using `ClosureReason` and form the `ErrorMessage` at the call site.

Reply:
Hmm, that would make it rather hard to return the error message. In most cases it looks like there's a 1-to-1 mapping between `ClosureReason` and the `ErrorMessage` message, but we'd have to do that mapping on the `ChannelManager` side, and lose a bit of info here or there. I'm not sure if that's really all that much cleaner, but we could make the return value a struct if you prefer that?

Reply:
Couldn't you just make a method on `ClosureReason` to give the string?

Reply:
Ah, good point, yes, that was much cleaner than I was thinking it would be.